### PR TITLE
Fix for S3:

### PR DESCRIPTION
**Pull request by jdrossl**
- Support copy operation for large files

https://github.com/craftercms/craftercms/issues/5215

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v1/service/aws/AwsUtils.java [hunk 1]
```diff
@@ -29,12 +29,16 @@
 import java.util.LinkedList;
 import java.util.List;
 
+import static java.util.stream.Collectors.toList;
+
 public abstract class AwsUtils {
 
     private static final Logger logger = LoggerFactory.getLogger(AwsUtils.class);
 
     public static final int MIN_PART_SIZE = 5 * 1024 * 1024;
```
Comment (avasquez614) [inline src/main/java/org/craftercms/studio/impl/v1/service/aws/AwsUtils.java:38]: This constant seems to not be used

File: src/main/java/org/craftercms/studio/impl/v1/service/aws/AwsUtils.java [hunk 2]
```diff
@@ -29,12 +29,16 @@
...
 
+    public static final int COPY_PART_SIZE = 1024 * 1024 * 1024;
```
Comment (avasquez614) [inline src/main/java/org/craftercms/studio/impl/v1/service/aws/AwsUtils.java:40]: This constant seems to not be used

File: src/main/java/org/craftercms/studio/impl/v1/service/aws/AwsUtils.java [hunk 3]
```diff
@@ -97,6 +101,65 @@ public static void uploadStream(String inputBucket, String inputKey, AmazonS3 s3
...
     }
 
+    public static void copyFile(String sourceBucket, String sourceKey, String destBucket, String destKey,
+                                int partSize, AmazonS3 client) {
+        logger.debug("Copying file from {}/{} to {}/{}", sourceBucket, sourceKey, destBucket, destKey);
+        GetObjectMetadataRequest metadataRequest = new GetObjectMetadataRequest(sourceBucket, sourceKey);
+        ObjectMetadata metadataResult = client.getObjectMetadata(metadataRequest);
+        long objectSize = metadataResult.getContentLength();
+
+        if (objectSize >= 5L * 1024L * 1024L * 1024L) {
```
Comment (avasquez614) [inline src/main/java/org/craftercms/studio/impl/v1/service/aws/AwsUtils.java]: Shouldn't this use a constant?

**Review by avasquez614 — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/impl/v1/service/aws/AwsUtils.java (hunk 1 above)
Comment (jdrossl) [reply src/main/java/org/craftercms/studio/impl/v1/service/aws/AwsUtils.java:38]: it is used in other classes

File: src/main/java/org/craftercms/studio/impl/v1/service/aws/AwsUtils.java (hunk 2 above)
Comment (jdrossl) [reply src/main/java/org/craftercms/studio/impl/v1/service/aws/AwsUtils.java:40]: it is used in other classes

**Review by avasquez614 — APPROVED** [review]

**Pull request merged**